src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,227 @@
+              + "when the trackKnownUsers setting is enabled.");
+      return;
+    }
+
+    String id = track.anonymousId();
+    String event = track.event();
+    Map<String, String> properties = track.properties().toStringMap();
+
+    if (trackKnownUsers && !isNullOrEmpty(userId)) {
+      id = track.userId();
+    }
+
+    client.track(event, id, attributes, properties);
+    logger.verbose("client.track(%s, %s, %s, %s)", event, id, attributes, properties);
+  }
+
+  @Override
+  public void reset() {
+    super.reset();
+
+    client.removeNotificationListener(listener);
+    logger.verbose("client.removeNotificationListener(listener)");
+  }
+
+  private void pollOptimizelyClient(final OptimizelyManager manager, final Analytics analytics) {
+    final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+
+    final Runnable poll =
+        new Runnable() {
+          @Override
+          public void run() {
+            synchronized (OptimizelyXIntegration.this) {
+              OptimizelyXIntegration.this.client = manager.getOptimizely();
+            }
+            if (client.isValid()) {
+              handler.post(
+                  new Runnable() {
+                    @Override
+                    public void run() {
+                      setClientAndFlushTracks(analytics);
+                    }
+                  });
+              scheduler.shutdown();
+            }
+          }
+        };
+    scheduler.scheduleAtFixedRate(poll, 60, 60, SECONDS);
+  }
+
+  private void setClientAndFlushTracks(Analytics analytics) {
+    this.listener = createListener(analytics);
+    client.addNotificationListener(listener);
+    logger.verbose("Flushing track queue");
+
+    for (TrackPayload t : trackEvents) {
+      track(t);
+    }
+
+    trackEvents = null;
   }
 
-  @Override public void identify(IdentifyPayload identify) {
+  private OptimizelyNotificationListener createListener(Analytics analytics) {
+    return new OptimizelyNotificationListener(analytics);
   }
 
-  @Override public void track(TrackPayload track) {
+  static class OptimizelyNotificationListener extends NotificationListener {
+    private final Analytics analytics;
+
+    public OptimizelyNotificationListener(Analytics analytics) {
+      this.analytics = analytics;
+    }
+
+    @Override
+    public void onExperimentActivated(
+        Experiment experiment, String userId, Map<String, String> attributes, Variation variation) {
+
+      Properties properties =
+          new Properties()
+              .putValue("experimentId", experiment.getId())
+              .putValue("experimentName", experiment.getKey())
+              .putValue("variationId", variation.getId())
+              .putValue("variationName", variation.getKey())
+              .putValue("nonInteraction", 1);
+
+      if (!nonInteraction) {
+        properties.remove("nonInteraction");
+      }
+      analytics.track("Experiment Viewed", properties, options);
```

Comment:
I think you will also need to pass in `Optimizely X : no` so the event does not attempt to go back into Optimizely

Reply:
Good catch! That's already specified in the `options` object passed into the event.